src/modules/SystemSettings.rb

```diff
@@ -15,210 +15,304 @@ module Yast
+    def valid_scheduler?(elevator)
+      GetPossibleElevatorValues().include?(elevator)
+    end
 
-    def Write
-      # writing SysRq settings
-      if @ENABLE_SYSRQ != nil && Builtins.regexpmatch(@ENABLE_SYSRQ, "^[0-9]+$")
-        # save the SysRq setting
-        Builtins.y2milestone("Saving ENABLE_SYSRQ: %1", @ENABLE_SYSRQ)
-        SCR.Write(path(".etc.sysctl_conf.\"kernel.sysrq\""), @ENABLE_SYSRQ)
-        SCR.Write(path(".etc.sysctl_conf"), nil)
-      else
-        Builtins.y2warning(
-          "Not writing invalid ENABLE_SYSRQ value: %1",
-          @ENABLE_SYSRQ
+    # Determine the current scheduler from the system
+    #
+    # @return [String] IO Scheduler name; if it's not valid/set, it will return an empty string
+    def current_elevator
+      # get 'elevator' option from the default section
+      elevator_parameter = Bootloader.kernel_param(:common, "elevator")
+      log.info("elevator_parameter: #{elevator_parameter}")
+
+      if elevator_parameter == :missing    # Variable is not set
+        ""
+      elsif elevator_parameter == :present # Variable is set but has not parameter
+        log.info("'elevator' variable has to have some value")
+        ""
+      elsif !valid_scheduler?(elevator_parameter.to_s) # Variable is set but hasn't any known value
+        log.warn(
+          format("'elevator' variable has to have a value from %s instead of being set to %s",
+            GetPossibleElevatorValues(),
+            elevator_parameter
+          )
         )
+        ""
+      else
+        elevator_parameter.to_s
       end
+    end
 
-      # the bootloader configuration is written at the end of the first stage
-      if Mode.normal
-        # write the elevator setting
-        Yast.import "Bootloader"
-        Bootloader.Write
+    # Activate SysRq keys configuration
+    #
+    # @see enable_sysrq
+    def activate_sysrq
+      if enable_sysrq.nil? || enable_sysrq !~ /\A[0-9]\z/
+        log.warn("Not activating invalid ENABLE_SYSRQ value: #{enable_sysrq}")
+        return
       end
 
-      true
+      log.info("Activating SysRq config: #{enable_sysrq}")
+      SCR.Execute(path(".target.bash"), "echo '#{enable_sysrq}' > /proc/sys/kernel/sysrq")
     end
 
-    # Kernel param 'elevator'
-    def GetIOScheduler
-      @elevator
+    # Activate IO scheduler
+    #
+    # @see activate_scheduler
+    def activate_scheduler
+      return if GetIOScheduler().nil?
+
+      Yast.import "Bootloader"
+      new_elevator = GetIOScheduler() == "" ? :missing : GetIOScheduler()
+
+      log.info("Activating scheduler: #{new_elevator}")
+      # set the scheduler
+      Bootloader.modify_kernel_params("elevator" => new_elevator)
+      # set bootloader configuration as 'changed' (bsc#968192)
+      Bootloader.proposed_cfg_changed = true
+
+      # activate the scheduler for all disk devices
+      return if new_elevator == :missing
+      Dir["/sys/block/*/queue/scheduler"].each do |f|
```

Comment:
AFAIK, the old code did, so I kept it. I guess that we could take advantage of the new scheduler.